pkg/cluster/internal/kubeadm/config.go

```diff
@@ -536,7 +658,7 @@ func Config(data ConfigData) (config string, err error) {
 	}
 
 	// assume the latest API version, then fallback if the k8s version is too low
-	templateSource := ConfigTemplateBetaV2
+	templateSource := ConfigTemplateAutoDiscoveryBetaV2
```

Comment:
```patch
--- 1.txt	2020-04-25 23:27:01.317131100 +0300
+++ 2.txt	2020-04-25 23:26:58.335960600 +0300
@@ -1,4 +1,4 @@
-const ConfigTemplateBetaV2 = `# config generated by kind
+const ConfigTemplateAutoDiscoveryBetaV2 = `# config generated by kind
 apiVersion: kubeadm.k8s.io/v1beta2
 kind: ClusterConfiguration
 metadata:
@@ -11,8 +11,11 @@
 # to the cluster after rewriting the kubeconfig to point to localhost
 apiServer:
   certSANs: [localhost, "{{.APIServerAddress}}"]
-{{ if .FeatureGates }}
   extraArgs:
+    {{ if .IPv6 -}}
+    bind-address: "::"
+    {{- end }}
+{{ if .FeatureGates }}
     "feature-gates": "{{ .FeatureGatesString }}"
 {{ end }}
 controllerManager:
@@ -46,16 +49,17 @@
 # we use a well know token for TLS bootstrap
 bootstrapTokens:
 - token: "{{ .Token }}"
-# we use a well know port for making the API server discoverable inside docker network. 
+# we use a well know port for making the API server discoverable inside docker network.
 # from the host machine such port will be accessible via a random local port instead.
 localAPIEndpoint:
-  advertiseAddress: "{{ .NodeAddress }}"
   bindPort: {{.APIBindPort}}
 nodeRegistration:
   criSocket: "/run/containerd/containerd.sock"
   kubeletExtraArgs:
     fail-swap-on: "false"
-    node-ip: "{{ .NodeAddress }}"
+    {{ if .IPv6 -}}
+    node-ip: "::"
+    {{- end }}
 ---
 # no-op entry that exists solely so it can be patched
 apiVersion: kubeadm.k8s.io/v1beta2
@@ -65,14 +69,15 @@
 {{ if .ControlPlane -}}
 controlPlane:
   localAPIEndpoint:
-    advertiseAddress: "{{ .NodeAddress }}"
     bindPort: {{.APIBindPort}}
 {{- end }}
 nodeRegistration:
   criSocket: "/run/containerd/containerd.sock"
   kubeletExtraArgs:
     fail-swap-on: "false"
-    node-ip: "{{ .NodeAddress }}"
+    {{ if .IPv6 -}}
+    node-ip: "::"
+    {{- end }}
 discovery:
   bootstrapToken:
     apiServerEndpoint: "{{ .ControlPlaneEndpoint }}"
```

wondering why my diff is picking a change on the `we use a well know port for making...` line though
/lgtm